code/mod_table/mod_table.cpp

```diff
@@ -380,16 +382,33 @@ void parse_mod_table(const char *filename)
 
 		if (optional_string("$Generic Pain Flash Factor:")) {
 			stuff_float(&Generic_pain_flash_factor);
-			if (Generic_pain_flash_factor != 1.0f)
+			if (!fl_equal(Generic_pain_flash_factor, 1.0f))
```

Comment:
Come to think of it, why is this being compared to 1 anyway?  The shield pain and EMP pain flash factors are only set if the factor is nonzero.

(And, bleh, the parser should make sure all three are positive too.)